### PR TITLE
Pi1 devices 2

### PR DESCRIPTION
**Pull request by elkcert**
Implements DUS1 and DMS. Might have misunderstood how DMS is supposed to work?

### CODE REVIEW

File: sensors/uds.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import RPi.GPIO as GPIO
+import time
+import typing
+import enum
+import random
+
+
+WAKEUP = 0.2
+MISC = 0.00001  # FIXME: What is this delay for?
```
Comment (magley) [inline sensors/uds.py]: To emit a sound wave, the HC-SR04 needs to receive a 5V signal for at least 10µs, or 0.00001s.

File: sensors/uds.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    iter = 0
+    while GPIO.input(pin_echo) == 1:
+        if iter > max_iter:
+            return UDSReading(UDSCode.TIMED_OUT, 0)
+        pulse_end_time = time.time()
+        iter += 1
+
+    pulse_duration = pulse_end_time - pulse_start_time
+    distance = (pulse_duration * 34300)/2
```
Comment (magley) [inline sensors/uds.py]: Consider making `34300` a constant (speed of sound)? We won't use it anywhere else so maybe it's okay to leave it here.

File: sensors/uds.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    return UDSReading(UDSCode.OK, distance)
+
+
+def read_simulator():
+    return UDSReading(UDSCode.OK, random.randint(10000, 20000))
```
Comment (magley) [inline sensors/uds.py]: Is this in mm? `random.randint(2, 400)` would be most accurate for the UDS (in cm).

File: sensors/uds.py (hunk 1 above)
Comment (elkcert) [reply sensors/uds.py]: https://github.com/magley/my-pi-home/pull/3/commits/eadbc344b96336e39034cd384cd573b99628ba5b

File: sensors/uds.py (hunk 2 above)
Comment (elkcert) [reply sensors/uds.py]: https://github.com/magley/my-pi-home/pull/3/commits/eadbc344b96336e39034cd384cd573b99628ba5b

File: sensors/uds.py (hunk 3 above)
Comment (elkcert) [reply sensors/uds.py]: https://github.com/magley/my-pi-home/pull/3/commits/a74878b017f8dcdf6b8a4156452458b42ce72584

**Review by magley — APPROVED** [review]
Looks okay. I see no issue with the keypad code, but it's hard to tell just by guessing.

**Pull request merged**